sla_per_user_system_test.py

```diff
@@ -622,3 +685,99 @@ def _throughput_latency_tests_run(self, read_cmds, read_users, latency_user):
+                interactive = float(workloads_results["interactive"][item])
+                batch = float(workloads_results["batch"][item])
+                ratio = interactive / batch if item == "op rate" else batch / interactive
+
+                comparison_results.update(
+                    {
+                        item: {
+                            "interactive": interactive,
+                            "batch": batch,
+                            "diff": batch - interactive,
+                            "ratio": ratio,
+                            "within_margin": ratio >= target_margin
+                        }
+                    }
+                )
+            return comparison_results
+        except Exception:
+            logger.info("Failed to compare c-s results for batch and interactive"
+                        "workloads.")
+            raise
+
+    def upload_c_s_comparison_to_es(self) -> None:
+        logger.info("Uploading c-s comparison to ES...")
+        es_body = {
+            self._get_db_node_instance_type(): {
+                "test_id": self.test_id,
+                "backend": self.db_cluster.params.get("cluster_backend"),
+                "scylla_version": self.get_scylla_versions(),
+                **self._comparison_results
+            }
+        }
+        self._es.create_doc(index="workload_types", doc_type="test_stats",
+                            doc_id=self.test_id, body=es_body)
+        logger.info("C-s comparison uploaded to ES.")
+
+    def _get_db_node_instance_type(self):
+        backend = self.db_cluster.params.get("cluster_backend")
+        if backend in ("aws", "aws-siren"):
+            return self.db_cluster.params.get("instance_type_db")
+        elif backend in ("gce", "gce-siren"):
+            return self.db_cluster.params.get("gce_instance_type_db")
+        else:
+            logger.warning("Unrecognized backend type, defaulting to 'Unknown' for"
+                           "db instance type.")
+            return None
+
+    def get_email_data(self):
+        self.log.info("Prepare data for email")
+        email_data = {}
+        grafana_dataset = {}
+
+        try:
+            email_data = self._get_common_email_data()
+        except Exception as error:  # pylint: disable=broad-except
+            self.log.error("Error in gathering common email data: Error:\n%s", error)
+
+        try:
+            grafana_dataset = self.monitors.get_grafana_screenshot_and_snapshot(
+                self.start_time) if self.monitors else {}
+        except Exception as error:  # pylint: disable=broad-except
+            self.log.error("Error in gathering Grafana screenshots and snapshots. Error:\n%s", error)
+
+        email_data.update({"grafana_screenshots": grafana_dataset.get("screenshots", []),
+                           "grafana_snapshots": grafana_dataset.get("snapshots", []),
+                           "scylla_ami_id": self.params.get("ami_id_db_scylla") or "-",
```

Comment:
if the backend is GCE?

Reply:
This will default to "-". Not sure we're going to run this on GCE for the forseeable future though.